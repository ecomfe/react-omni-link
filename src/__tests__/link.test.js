import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

import {Link, ExternalLink, HyperLink, createLink} from '../index';

test('passing external prop to <Link>', () => {
    const link = renderer
        .create(<Link external to="http://www.facebook.com">Facebook</Link>)
        .toJSON();
    expect(link).toMatchSnapshot();
});


test('passing refresh prop to <Link>', () => {
    const link = renderer
        .create(<Link refresh to="http://www.facebook.com">Facebook</Link>)
        .toJSON();
    expect(link).toMatchSnapshot();
});


test('passing activeClassName prop to <Link>', () => {
    const link = renderer
        .create(
            <BrowserRouter>
                <div>
                    <Link activeClassName="active" to="/">Home</Link>
                </div>
            </BrowserRouter>
        )
        .toJSON();
    expect(link).toMatchSnapshot();
});

test('<ExternalLink>', () => {
    const link = renderer
        .create(<ExternalLink to="http://www.facebook.com">Facebook</ExternalLink>)
        .toJSON();
    expect(link).toMatchSnapshot();
});

test('<HyperLink>', () => {
    const link = renderer
        .create(<HyperLink to="http://www.facebook.com">Facebook</HyperLink>)
        .toJSON();
    expect(link).toMatchSnapshot();
});

test('react-router using <Link>', () => {
    const link = renderer
        .create(
            <BrowserRouter>
                <div>
                    <Link to="/">Home</Link>
                </div>
            </BrowserRouter>
        )
        .toJSON();
    expect(link).toMatchSnapshot();
});

test('createLink', () => {
    const MyLink = createLink('/users');
    const link = renderer
        .create(
            <BrowserRouter>
                <div>
                    <MyLink>Home</MyLink>
                </div>
            </BrowserRouter>
        )
        .toJSON();
    expect(link).toMatchSnapshot();
})

test('createLink with template', () => {
    const MyLink = createLink('/users/{id}');
    const link = renderer
        .create(
            <BrowserRouter>
                <div>
                    <MyLink id={1}>Home</MyLink>
                </div>
            </BrowserRouter>
        )
        .toJSON();
    expect(link).toMatchSnapshot();
})
