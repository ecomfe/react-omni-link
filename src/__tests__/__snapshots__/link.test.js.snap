// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExternalLink> 1`] = `
<a
  href="http://www.facebook.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Facebook
</a>
`;

exports[`<HyperLink> 1`] = `
<a
  href="http://www.facebook.com"
>
  Facebook
</a>
`;

exports[`createLink 1`] = `
<div>
  <a
    href="/users"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;

exports[`createLink with template 1`] = `
<div>
  <a
    href="/users/1"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;

exports[`passing activeClassName prop to <Link> 1`] = `
<div>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Home
  </a>
</div>
`;

exports[`passing external prop to <Link> 1`] = `
<a
  href="http://www.facebook.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Facebook
</a>
`;

exports[`passing refresh prop to <Link> 1`] = `
<a
  href="http://www.facebook.com"
>
  Facebook
</a>
`;

exports[`react-router using <Link> 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;
